### PR TITLE
OPENEUROPA-3331: Fix document rendering.

### PR DESCRIPTION
**Pull request by imanoleguskiza**
### Description

Fixed document not showing the appropriate language and use translated document pattern.

### Change log

- Added:
- Changed: Use translated document pattern.
- Deprecated:
- Removed:
- Fixed: Document not showing correct language.
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: tests/Kernel/MediaRenderTest.php [hunk 1]
```diff
@@ -59,44 +61,101 @@ protected function setUp() {
...
+        $translation_file_info_language = $translation_list->filter('.ecl-file__translation-item .ecl-file__translation-info div.ecl-file__translation-title');
+        $this->assertContains($language_names[$translation_language->getId()][1], $translation_file_info_language->text());
+
+        $translation_file_info_properties = $translation_list->filter('.ecl-file__translation-item .ecl-file__translation-info div.ecl-file__translation-meta');
+        $this->assertContains('KB - PDF)', $translation_file_info_properties->text());
+
+        $translation_file_download_link = $translation_list->filter('.ecl-file__translation-download');
+        $this->assertContains('/test_' . $translation_language->getId() . '.pdf', $translation_file_download_link->attr('href'));
+        $this->assertContains('Download', $file_download_link->text());
+      }
```
Comment (22Alexandra) [inline tests/Kernel/MediaRenderTest.php:150]: Test also for a document without translations.

**Review by 22Alexandra — CHANGES_REQUESTED** [review]

File: tests/Kernel/MediaRenderTest.php (hunk 1 above)
Comment (imanoleguskiza) [reply tests/Kernel/MediaRenderTest.php:150]: Good point.

**Review by 22Alexandra — DISMISSED** [review]

File: tests/Kernel/MediaRenderTest.php [hunk 2]
```diff
@@ -59,45 +62,133 @@ protected function setUp() {
    * Tests that the Document media type is rendered with the correct ECL markup.
    */
   public function testDocumentMedia(): void {
-    $file = file_save_data(file_get_contents(drupal_get_path('module', 'oe_media') . '/tests/fixtures/sample.pdf'), 'public://test.pdf');
-    $file->setPermanent();
-    $file->save();
+    // Make image media translatable.
+    $setting = $this->container->get('entity_type.manager')->getStorage('language_content_settings')->create([
+      'langcode' => 'en',
+      'statur' => TRUE,
```
Comment (nagyad) [inline tests/Kernel/MediaRenderTest.php]: Typo.

File: tests/Kernel/MediaRenderTest.php [hunk 3]
```diff
@@ -59,45 +62,133 @@ protected function setUp() {
...
+      'id' => 'media.document',
+      'target_entity_type_id' => 'media',
+      'target_bundle' => 'document',
+      'default_langcode' => 'site_default',
+      'language_alterable' => 'true',
```
Comment (nagyad) [inline tests/Kernel/MediaRenderTest.php]: Isn't it a boolean?

File: tests/Kernel/MediaRenderTest.php [hunk 4]
```diff
@@ -59,45 +62,133 @@ protected function setUp() {
...
+      'oe_media_file' => [
+        'target_id' => $spanish_file->id(),
+      ],
+    ]);
+    $media_spanish->save();
 
-      $file_title = $file_row->filter('.ecl-file__title');
-      $this->assertContains('test document', $file_title->text());
+    // Assert that the media is rendered properly with translations in
+    // all available languages.
```
Comment (nagyad) [inline tests/Kernel/MediaRenderTest.php]: I interpret this sentence as you are checking it in all 24 languages available on the site and I am not sure it is really what you do here. I think you are checking for all available translations or translation languages.

File: tests/Kernel/MediaRenderTest.php [hunk 5]
```diff
@@ -59,45 +62,133 @@ protected function setUp() {
...
+    $translation_languages = $media->getTranslationLanguages();
+    foreach (['default', 'oe_theme_main_content'] as $view_mode) {
+      foreach ($translation_languages as $document_langcode => $document_language) {
+        $build = $this->mediaViewBuilder->view($media, $view_mode, $document_langcode);
+        $crawler = new Crawler($this->renderRoot($build));
+
+        // Check the rendering of the current language document.
+        $this->assertMainDocumentRendering($crawler, $document_language);
+        // Check the rendering of the available translations.
+        // File translation list.
```
Comment (nagyad) [inline tests/Kernel/MediaRenderTest.php]: This is not required.

File: tests/Kernel/MediaRenderTest.php [hunk 6]
```diff
@@ -59,45 +62,133 @@ protected function setUp() {
...
+
+        // Get the available translation language.
+        $translation_language = array_filter($translation_languages, function ($langcode) use ($document_langcode) {
+          return $document_langcode != $langcode;
+        }, ARRAY_FILTER_USE_KEY);
+        /** @var \Drupal\Core\Language\Language $translation_language */
+        $translation_language = reset($translation_language);
+
+        $language_names = $predefined = EuropeanUnionLanguages::getLanguageList() + LanguageManager::getStandardLanguageList();
+        $translation_file_info_language = $translation_list->filter('.ecl-file__translation-item .ecl-file__translation-info div.ecl-file__translation-title');
```
Comment (nagyad) [inline tests/Kernel/MediaRenderTest.php]: I think this should be $file_translation_info_... It is not a translation file.

**Review by nagyad — CHANGES_REQUESTED** [review]

**Comment by nagyad** [comment]
image
Can you please check with Tamas if this should be a link?

**Comment by imanoleguskiza** [comment]
Checked with Tamas and agreed to remove it for the time being.

File: tests/Kernel/MediaRenderTest.php [hunk 7]
```diff
@@ -59,45 +62,132 @@ protected function setUp() {
...
+    $english_file->setPermanent();
+    $english_file->save();
+
+    // Create spanish file.
+    $spanish_file = file_save_data(file_get_contents(drupal_get_path('module', 'oe_media') . '/tests/fixtures/sample.pdf'), 'public://test_es.pdf');
+    $spanish_file->setPermanent();
+    $spanish_file->save();
 
+    // View modes to test.
+    $view_modes = ['default', 'oe_theme_main_content'];
```
Comment (nagyad) [inline tests/Kernel/MediaRenderTest.php]: Let's move this part before the foreach.

File: tests/Kernel/MediaRenderTest.php [hunk 8]
```diff
@@ -59,45 +62,132 @@ protected function setUp() {
...
+      $this->assertMainDocumentRendering($crawler, $media->language());
+      // Assert no translations are rendered.
+      $translation_list = $crawler->filter('.ecl-file__translation-list');
+      $this->assertCount(0, $translation_list);
+    }
 
-      // File row.
-      $file_row = $crawler->filter('.ecl-file .ecl-file__container');
-      $this->assertCount(1, $file_row);
+    // Translate the media to spanish.
```
Comment (nagyad) [inline tests/Kernel/MediaRenderTest.php]: Spanish is capital S in English.

File: tests/Kernel/MediaRenderTest.php [hunk 9]
```diff
@@ -59,45 +62,132 @@ protected function setUp() {
...
+      ],
+    ]);
+    $media_spanish->save();
 
-      $file_title = $file_row->filter('.ecl-file__title');
-      $this->assertContains('test document', $file_title->text());
+    // Assert that the media and its translations are rendered properly
+    // in all translated languages.
+    $translation_languages = $media->getTranslationLanguages();
+    foreach (['default', 'oe_theme_main_content'] as $view_mode) {
```
Comment (nagyad) [inline tests/Kernel/MediaRenderTest.php]: Let's use here the $view_modes variable.

File: tests/Kernel/MediaRenderTest.php [hunk 10]
```diff
@@ -59,45 +62,132 @@ protected function setUp() {
...
+      'langcode' => 'en',
+      'status' => TRUE,
+      'id' => 'media.document',
+      'target_entity_type_id' => 'media',
+      'target_bundle' => 'document',
+      'default_langcode' => 'site_default',
+      'language_alterable' => TRUE,
+    ]);
+    $setting->setThirdPartySetting('content_translation', 'enabled', TRUE);
+    $setting->save();
```
Comment (nagyad) [inline tests/Kernel/MediaRenderTest.php]: I believe you could make this just in two lines:
`$this->container->get('content_translation.manager')->setEnabled('media', 'document', TRUE);`
` $this->container->get('router.builder')->rebuild();`

File: tests/Kernel/MediaRenderTest.php [hunk 11]
```diff
@@ -59,45 +62,132 @@ protected function setUp() {
...
+        $this->assertContains('/test_' . $translation_language->getId() . '.pdf', $translation_file_download_link->attr('href'));
+        $this->assertContains('Download', $translation_file_download_link->text());
+      }
+    }
+  }
 
-      $file_info_language = $file_row->filter('.ecl-file__info div.ecl-file__language');
-      $this->assertContains('English', $file_info_language->text());
+  /**
+   * Helper test to assert the rendering of the main document.
```
Comment (nagyad) [inline tests/Kernel/MediaRenderTest.php]: Assert the rendering of the main document. - is sufficient.

**Review by nagyad — CHANGES_REQUESTED** [review]

File: tests/Kernel/MediaRenderTest.php [hunk 12]
```diff
@@ -59,45 +62,123 @@ protected function setUp() {
...
   public function testDocumentMedia(): void {
-    $file = file_save_data(file_get_contents(drupal_get_path('module', 'oe_media') . '/tests/fixtures/sample.pdf'), 'public://test.pdf');
-    $file->setPermanent();
-    $file->save();
-
+    // Make image media translatable.
+    $this->container->get('content_translation.manager')->setEnabled('media', 'document', TRUE);
+    $this->container->get('router.builder')->rebuild();
+
+    // Create english file.
```
Comment (nagyad) [inline tests/Kernel/MediaRenderTest.php]: forgot English.

File: tests/Kernel/MediaRenderTest.php (hunk 8 above)
Comment (nagyad) [reply tests/Kernel/MediaRenderTest.php:106]: !

**Review by nagyad — CHANGES_REQUESTED** [review]

**Review by nagyad — DISMISSED** [review]

File: oe_theme.theme [hunk 13]
```diff
@@ -1206,12 +1206,25 @@ function oe_theme_preprocess_media__document__default(&$variables) {
...
+    ->setTitle($media->getName())
+    ->setLanguageCode($media->language()->getId());
+  // Generate the file information for all available translations.
+  foreach ($media->getTranslationLanguages() as $langcode => $language) {
+    // We don't want to include the information of the current language again.
+    if ($media->language()->getId() === $langcode) {
+      continue;
+    }
+    $translation = $media->getTranslation($langcode);
+    $file_object = $translation->get('oe_media_file')->first()->entity;
```
Comment (sergepavle) [inline oe_theme.theme:1224]: Looks like we have to do checking that entity is an instance of fileEntityInterface (as Drupal core doing often)

File: tests/Kernel/MediaRenderTest.php [hunk 14]
```diff
@@ -59,45 +62,123 @@ protected function setUp() {
...
       ],
     ]);
 
     $media->save();
 
-    foreach (['default', 'oe_theme_main_content'] as $view_mode) {
+    // View modes to test.
+    $view_modes = ['default', 'oe_theme_main_content'];
+
+    // Assert tha the media is rendered properly without translations.
```
Comment (sergepavle) [inline tests/Kernel/MediaRenderTest.php]: typo

**Review by sergepavle — CHANGES_REQUESTED** [review]
Minor comments

**Review by sergepavle — APPROVED** [review]

**Review by ademarco — APPROVED** [review]

**Pull request merged**